### PR TITLE
consensus: Replace CypressCreditV2 at KaiaForkBlockParent

### PR DESCRIPTION
**Pull request by blukat29**
## Proposed changes

- Replace `CypressCredit.sol { getPhoto, getNames }` with `CypressCreditV2.sol { getPhoto, getNames, getEndingPhoto, getEndingNames }` at the KaiaForkBlockParent.
- Update `klay_getCypressCredit` API to return `{ photo, names, endingPhoto?, endingNames? }`
- The EndingPhoto and EndingNames to be updated as a separate PR.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [ ] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [ ] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [ ] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

Tested with homi.

```
homi setup --gen-type local --cn-num 1 --kaia-compatible-blocknumber 60 --cypress --chainID 8217
```
```
INFO[05/15,14:51:28 +09] [24|consensus/istanbul/backend/engine.go:575]   Replaced CypressCredit with CypressCreditV2  blockNum=59
```
```
> var out = klay.getCypressCredit(); console.log(out.names, '//', out.endingNames, '@', klay.blockNumber);
 Andrew Kim, ..., Zachary Keats //  @ 6
> var out = klay.getCypressCredit(); console.log(out.names, '//', out.endingNames, '@', klay.blockNumber);
 Andrew Kim, ..., Zachary Keats // EndingNames TBU @ 65
```

### CODE REVIEW

File: contracts/contracts/system_contracts/misc/CypressCreditV2.sol [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+pragma solidity 0.4.24;
+
+import "./CypressCredit.sol";
+
+/// @title Cypress Genesis and Ending Block Information
+
+// CypressCreditV2 keeps the information about Klaytn team when this contract is
+// created, and it replaces the CypressCredit contract in the Klaytn Cypress mainnet.
+contract CypressCreditV2 is CypressCredit {
+    // getPhoto returns a base64-encoded photo of the Klaytn team.
```
Comment (hyeonLewis) [inline contracts/contracts/system_contracts/misc/CypressCreditV2.sol:26]:
```suggestion
    // getEndingPhoto returns a base64-encoded photo of the Klaytn team.
```

File: contracts/contracts/system_contracts/misc/CypressCreditV2.sol [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    function getEndingPhoto() public pure returns (string) {
+        string memory photo = "EndingPhoto TBU";
+        return photo;
+    }
+
+    // getNames returns comma-separated names of all the members in the Klaytn team.
```
Comment (hyeonLewis) [inline contracts/contracts/system_contracts/misc/CypressCreditV2.sol:32]:
```suggestion
    // getEndingNames returns comma-separated names of all the members in the Klaytn team.
```

File: contracts/contracts/system_contracts/misc/CypressCreditV2.sol [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+pragma solidity 0.4.24;
+
+import "./CypressCredit.sol";
+
+/// @title Cypress Genesis and Ending Block Information
+
+// CypressCreditV2 keeps the information about Klaytn team when this contract is
+// created, and it replaces the CypressCredit contract in the Klaytn Cypress mainnet.
+contract CypressCreditV2 is CypressCredit {
```
Comment (hyeonLewis) [inline contracts/contracts/system_contracts/misc/CypressCreditV2.sol:25]: Just for rhyming; What about overriding the V1 functions like:
```
function getStartPhoto() public pure returns (string) {
    return getPhoto();
}
// Same for names.
```

File: contracts/contracts/system_contracts/misc/CypressCreditV2.sol [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        string memory photo = "EndingPhoto TBU";
+        return photo;
+    }
+
+    // getNames returns comma-separated names of all the members in the Klaytn team.
+    function getEndingNames() public pure returns (string) {
+        string memory names = "EndingNames TBU";
+        return names;
+    }
+}
```
Comment (hyeonLewis) [inline contracts/contracts/system_contracts/misc/CypressCreditV2.sol:37]: It's just a question: Don't we need to make it possible to receive KAIA by adding `receive`?

Comment (blukat29) [reply contracts/contracts/system_contracts/misc/CypressCreditV2.sol:37]: Let's discuss when we fill in the TBUs above.

**Review by hyeonLewis — DISMISSED** [review]

**Review by ian0371 — APPROVED** [review]

**Review by hyeonLewis — APPROVED** [review]

**Review by 2dvorak — APPROVED** [review]

**Pull request merged**